definitions/features/service.rb

```diff
@@ -44,8 +44,18 @@ def action_past_tense(action)
   end
 
   def filter_disabled_services!(action, service_list)
+    enabled_services = []
     if %w[start stop restart status].include?(action)
-      service_list.select!(&:enabled?)
+      service_list.each do |service|
+        # External db services can't be enabled or disabled
+        # Hence running the filteration on local services only
+        if service.respond_to?(:enabled?)
+          enabled_services << service if service.enabled?
+        else
+          enabled_services << service
+        end
+      end
+      return enabled_services
     end
     service_list
```

Comment:
This does change the behavior. If the action isn't in the included list, it now returns an empty list. A possible smaller diff that maintains the current behavior:
```ruby
   def filter_disabled_services!(action, service_list) 
    if %w[start stop restart status].include?(action)
      service_list.select! { |service| !service.respond_to?(:enabled?) || service.enabled? }
    end
    service_list
  end
```

Reply:
Yes to avoid returning empty list I explicitly returned the `enabled_services` from if block. Your approach if far cleaner, thank you!